### DISCUSSION TITLE
How to share after running a script?

### DISCUSSION THREAD

**Discussion by kzshantonu** [Waboodoo/HTTP-Shortcuts#327] [Q&A] [answered]
I have a script that takes input from share function and removes everything except the URL, I have it set to copy to clipboard, but would like to share it. Cannot seem to find a way:

```
const sharedValue = getVariable('text_and_url')
const matches = sharedValue.match(/\bhttps?:\/\/\S+/gi);
const url = matches[0];
copyToClipboard(url);
showToast(url);
```

Any help?

**Comment by Waboodoo** [comment, marked as answer]
The app currently doesn't have a feature to share text directly from Scripting, but you can work around this via the `sendIntent` function. To do so, copy the following code snippet:

```js
sendIntent({
  "type": "activity",
  "action": "android.intent.action.SEND",
  "dataType": "text/plain",
  "dataUri": " ",
  "extras": [
    {
      "name": "android.intent.extra.TEXT",
      "value": "Your text here"
    }
  ]
});
```

I will add a more convenient way to share text into the next version (2.30.0).

**Reply by Waboodoo** [reply]
As of version 2.30.0, you can now easily share text via the `shareText()` function.
See https://http-shortcuts.rmy.ch/scripting#share-text
